test/test_sparse.py

```diff
@@ -255,6 +255,20 @@ def test_tensor(x, res):
         res = self.ValueTensor(3, 4, 5, 0)
         test_tensor(x, res)
 
+    def test_to_sparse(self):
+        shape = [10, 5, 19, 8]
+        for dim, dim_sz in enumerate(shape):
+            sparse_dims = dim + 1
+            for nnz in range(1, dim_sz):
+                expected, _, _ = self._gen_sparse(sparse_dims, nnz, shape)
+                d = expected.to_dense()
+                result = d.to_sparse(sparse_dims)
+                self.assertEqual(expected.to_dense(), result.to_dense())  # == not implemented for sparse tensors yet
+                self.assertEqual(sparse_dims, result._sparseDims())
+
+        sp, _, _ = self._gen_sparse(2, 10, [3, 3, 3])
+        self.assertRaises(RuntimeError, lambda: sp.to_sparse())
```

Comment:
why is this an error?

Reply:
To me "to sparse" implies the source is not sparse.  are you looking for a sparse -> sparse routine that can alter the # of sparseDims (not inplace)?

Reply:
ah, I see, `S -> S` should error out, you are right!